src/components/app/create/CreateApp.tsx

```diff
@@ -47,159 +40,130 @@ export class AddNewApp extends Component<AddNewAppProps, AddNewAppState> {
+            typeof this.props.close === 'function'
+        ) {
+            this.props.close(evt)
+        }
+    }
 
     handleAppname(event: React.ChangeEvent<HTMLInputElement>): void {
-        let { form, isValid } = { ...this.state };
-        form.appName = event.target.value;
-        isValid.appName = this.rules.appName(event.target.value).isValid;
-        this.setState({ form, isValid, appNameErrors: true });
+        let { form, isValid } = { ...this.state }
+        form.appName = event.target.value
+        isValid.appName = this.rules.appName(event.target.value).isValid
+        this.setState({ form, isValid, appNameErrors: true })
     }
 
     handleProject(item: number): void {
-        let { form, isValid } = { ...this.state };
-        form.projectId = item;
-        isValid.projectId = !!item;
-        this.setState({ form, isValid });
+        let { form, isValid } = { ...this.state }
+        form.projectId = item
+        isValid.projectId = !!item
+        this.setState({ form, isValid })
     }
-    validateForm = (): boolean => {
-        if (
-            this.state.labels.tags.length !==
-            this.state.labels.tags.map((tag) => tag.value).filter((tag) => validateTags(tag)).length
-        ) {
-            this.setState({
-                labels: {
-                    ...this.state.labels,
-                    tagError: TAG_VALIDATION_MESSAGE.error,
-                },
-            });
-            return false;
-        }
-        return true;
-    };
 
-     getHostURLConfig = async () => {
-         try {
-             const { result } = await getHostURLConfiguration()
-             if (!result?.value) {
-                 const payload = {
-                     id: result?.id,
-                     key: result?.key || 'url',
-                     value: window.location.origin,
-                     active: result?.active || true,
-                 }
-                 await saveHostURLConfiguration(payload)
-             }
-         } catch (error) {
-             showError(error)
-         }
-     }
+    getHostURLConfig = async () => {
+        try {
+            const { result } = await getHostURLConfiguration()
+            if (!result?.value) {
+                const payload = {
+                    id: result?.id,
+                    key: result?.key || 'url',
+                    value: window.location.origin,
+                    active: result?.active || true,
+                }
+                await saveHostURLConfiguration(payload)
+            }
+        } catch (error) {
+            showError(error)
+        }
+    }
 
-    createApp(): void {
-        const validForm = this.validateForm();
-        if (!validForm) {
-            return;
+    createApp(e): void {
+        e.preventDefault()
+        const labelTags = []
+        let invalidLabels = false
+        for (let index = 0; index < this.state.tags.length; index++) {
+            const element = this.state.tags[index]
```

Comment:
Please do rename the element to some relatable name somewhat like `tagsIndex`

Reply:
Done